hip-0000-smart-contract-hcs-interoperable.md

```diff
@@ -0,0 +1,58 @@
+---
+hip: 478
+title: Interoperability Between Smart Contracts and the HCS
+author: John Conway (@scalemaildev), Walter Hernandez (@walter-hernandez)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2022-06-21T07:00:00Z
+created: 05-17-2022
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/479
+updated: 05-24-2022
+---
+
+## Abstract
+
+Enable smart contract interaction with the Hedera Consensus Service, by allowing contracts to create, read, and write to HCS topics.
+
+## Motivation
+
+Interoperability between the HCS and smart contracts would enable developers to leverage HCS-stored data in their dApps.
+
+## Rationale
+
+1. By enabling smart contracts to interact with the HCS, this would allow the creation of HCS-based Oracles. For example, a smart contract could subscribe to a private or public topic in order to read its external data, such as the prices of crypto-assets.
+
+2. This would also enable topic-based NFT minting rights. A smart contract could ingest an HCS topic to calculate the user's available NFTs. Then, after minting the NFT and transferring it to the user, the smart contract could update that same HCS topic. Currently, this process is only achievable via a centralized, server-based chain of events (such as on [https://hashandslash.io](hashandslash.io)).
+
+3. Any other Oracle use case could be repurposed by the HCS.
+
+## User stories
+
+As a developer, I want to be able to read HCS topics from my smart contracts.
+
+As a developer, I want to be able to send HCS topic messages from my smart contracts.
+
+As a developer, I want to be able to create HCS topics from my smart contracts.
+  
+## Specification
+
+Enabling smart contracts to interact with HCS topics would follow the same approach as the [https://hips.hedera.com/hip/hip-206](HTS pre-compiled contracts HIP).
```

Comment:
Like the noted HIP 206 can you call out the desired  pre-compile functions you're looking to be added, noting the input and output params in a similar fashion.
You can leave the hash empty to start with.

It would also be good to note any additional structs if you have any in mind.